### ISSUE TITLE
About implementation of inverse warp

### ISSUE THREAD

**Issue by haofeixu** [ClementPinard/SfmLearner-Pytorch#34]
Hi, @ClementPinard , I am a little confused by your implementation of [inverse warp function](https://github.com/ClementPinard/SfmLearner-Pytorch/blob/14ed93e346fbf9f81269fe6684760359cb157a2e/inverse_warp.py#L69) below:

```
if padding_mode == 'zeros':
        X_mask = ((X_norm > 1)+(X_norm < -1)).detach()
        X_norm[X_mask] = 2  # make sure that no point in warped image is a combinaison of im and gray
        Y_mask = ((Y_norm > 1)+(Y_norm < -1)).detach()
        Y_norm[Y_mask] = 2
```
 
for pixels out of range here, why do you set their value as 2?
It seems like these pixels will be padded by zeros, no matter their values are 2 or not, and what do you mean by your comment here? Could you please give some explanations? Thanks!

**Comment by haofeixu** [comment]
Anyone can help? Thanks!

**Comment by ClementPinard** [comment]
Hello, see here for an explanation :+1: https://discuss.pytorch.org/t/spatial-transformer-networks-boundary-grid-interpolation-behaviour/8891

TLDR : interpolation between image and void is considered differentiable and it can be a big problem for when the pixel is supposed to be outside

**Comment by haofeixu** [comment]
Thanks!
